src/targets/services/cleanOrphanAccounts.js

```diff
@@ -0,0 +1,81 @@
+const cozyFetch = require('../../lib/services/cozyFetch')
+
+const logMessagePrefix = level =>
+  `[cozy-collect][cleanOrphanAccounts][${level}]:`
+
+const output = level =>
+  function() {
+    console.log.apply(
+      console,
+      [logMessagePrefix(level)].concat(
+        // Node < 7 does not know about Object.values
+        Object.keys(arguments).map(key => arguments[key])
+      )
+    )
+  }
+
+const log = {
+  error: output('error'),
+  info: output('info')
+}
+
+const handleFetchError = error => log.error(error.message)
+
+log.info('Fetching accounts...')
+
+cozyFetch(
+  'GET',
+  '/data/io.cozy.accounts/_all_docs?include_docs=true',
+  null,
+  true
+)
+  .then(accounts => {
+    log.info(`${accounts.length || 0} account(s) has been found.`)
+    log.info('Fetching konnector triggers...')
+
+    cozyFetch('GET', '/jobs/triggers?Worker=konnector', null, true)
```

Comment:
It would be more readable, less error prone, and more easily testable

*  if the code was broken into small functions

fetchAccounts()
  .then(fetchKonnectors())
  .then(deleteOrphanAccounts())

* if the promises were flattened : http://blog.ministryofprogramming.com/prevent-pyramids-with-flat-promise-chains/

* By flattening the promises the error handler can be attached only one time

Reply:
I don't know if this repo has support for async/await but if it has, I think it should be a plus to use it.

Reply:
Yep, that was my first thought, but I need to have `accounts` all the way. `async/await` should be better, that was my first try but maybe I missed something.

Reply:
(The error handler can still be attached only one time, I had in mind to catch any fetch error, but you're right)

Reply:
For passing data to a lot of promises, you can declare the variable outside the
promise.

```js
const fetchAccounts = () => {
   return findAll('io.cozy.accounts')
}

const fetchTriggers = () => {
   return findAll('io.cozy.triggers')
}

const findOrphans = (accounts, triggers) => {
  const triggersByAccountId = indexBy(...)
  return accounts.filter(acc => !triggersByAccountId[acc._id]) 
}

const run = () => {
  let accounts, triggers
  Promise.all(() => {
    fetchAccounts.then(_accounts => accounts = _accounts),
    fetchTriggers.then(_triggers => triggers = _triggers)
  }).then(() => findOrphans(accounts, triggers))
    .then(orphanAccounts => deleteDocs('io.cozy.accounts', orphanAccounts))
}
```

Reply:
Here this technique is not even necessary since Promise.all would pass its results (accounts and triggers) directly to its callback.

Reply:
Thanks, actually I dislike using outside variable and try to do it the less as possible. I prefer using async/await, update incoming :)

Reply:
IMHO if you don't have access to async/await, the outside variable technique is a lesser evil than not breaking the code between functions.